ExchangeUnion/xud-docker
# feat: build from .src folder

Related to #590 

### What has been changed in this PR?

We are not using `git clone` in Dockerfile. We uses an external `images/<name>/.src` folder to hold all source code of each images. And in the future minor version upgrade will be easier by just triggering `tools/push <image>:<tag>` on xud-docker master branch in Travis-CI.

### How to test?

```
tools/build \
bitcoind:0.20.0 \
litecoind:0.18.1 \
geth:1.9.16 \
lnd:0.10.1-beta-ltc \
lnd:0.10.1-beta-ltc-simnet \
lnd:0.10.2-beta \
lnd:0.10.2-beta-simnet \
connext \
connext:7.0.0-alpha.14 \
xud \
xud:1.0.0-beta.5 \
arby \
arby:0.2.0 \
boltz \
boltz:1.0.0 \
webui \
webui:1.0.0
```

Run all 3 networks to see if anything breaks.

### Known issues

- [ ] The tools/build **modified images detection** is deprecated in this branch because this feature is error prone. You need to specify which images to build in `.travis.yaml` explicitly in script section like

```yaml
script: tools/push xud xud:1.0.0-beta.99 foo:bar
``` 

- [x] Travis 10 minutes no output error fix on the way
